app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyBottomSheet.kt

```diff
@@ -0,0 +1,131 @@
+        medicineFrequency: MedicineFrequencies
+    ): Intent {
+      return Intent(context, MedicineFrequencyBottomSheet::class.java).apply {
+        putExtra(DRUG_DURATION, drugDuration)
+        putExtra(MEDICINE_FREQUENCY, medicineFrequency)
+      }
+    }
+  }
+
+  private val drugDuration by unsafeLazy {
+    intent.getParcelableExtra<DrugDuration>(DRUG_DURATION)!!
+  }
+
+  private val medicineFrequency by unsafeLazy {
+    intent.getSerializableExtra(MEDICINE_FREQUENCY)
+  }
+
+  private val events by unsafeLazy {
+    saveClicks()
+  }
+
+  private val delegate by unsafeLazy {
+    MobiusDelegate.forActivity(
+        events = events.ofType(),
+        defaultModel = MedicineFrequencyModel.create(MedicineFrequencies.BD),
+        init = MedicineFrequencyInit(),
+        update = MedicineFrequencyUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build()
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.sheet_medicine_frequency)
+    medicineFrequencyTitleTextView.text = getString(R.string.drug_duration_title, drugDuration.name, drugDuration.dosage)
+  }
+
+  private fun saveClicks(): Observable<MedicineFrequencyEvent> {
+    return saveMedicineFrequencyButton
+        .clicks()
+        .map {
+          when {
+            medicineFrequencyOdRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.OD
+            medicineFrequencyBdRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.BD
+            medicineFrequencyTdsRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.TDS
+            else -> medicineFrequency == MedicineFrequencies.QDS
+          }
+          SaveMedicineFrequencyClicked(medicineFrequency as MedicineFrequencies)
+        }
+  }
+
+  override fun setMedicineFrequency(medicineFrequency: MedicineFrequencies) {
+    medicineFrequencyOdRadioButton.isChecked = medicineFrequency == MedicineFrequencies.OD
+    medicineFrequencyBdRadioButton.isChecked = medicineFrequency == MedicineFrequencies.BD
+    medicineFrequencyTdsRadioButton.isChecked = medicineFrequency == MedicineFrequencies.TDS
+    medicineFrequencyQdsRadioButton.isChecked = medicineFrequency == MedicineFrequencies.QDS
+  }
+
+  override fun saveMedicineFrequency(medicineFrequency: MedicineFrequencies) {
+    val intent = Intent().apply {
+      putExtra(SAVED_MEDICINE_FREQUENCY, medicineFrequency)
+    }
+    setResult(Activity.RESULT_OK, intent)
+    finish()
+  }
+
+  override fun onStart() {
+    super.onStart()
+    delegate.start()
+  }
+
+  override fun onStop() {
+    delegate.stop()
+    super.onStop()
+  }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    super.onSaveInstanceState(outState)
+    delegate.onSaveInstanceState(outState)
+  }
+
+  override fun attachBaseContext(newBase: Context?) {
+    setUpDependencyInjection()
+    super.attachBaseContext(newBase)
```

Comment:
We should also be updating the context here, please take a look at other activities.